PetAdoptions/pethistory/index.js

```diff
@@ -0,0 +1,237 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+const { Client } = require('pg');
+const Joi = require('joi');
+
+// Schema validation for SQS messages
+const adoptionMessageSchema = Joi.object({
+    transactionId: Joi.string().uuid().required(),
+    petId: Joi.string().required(),
+    petType: Joi.string().required(),
+    userId: Joi.string().required(),
+    adoptionDate: Joi.string().isoDate().required(),
+    timestamp: Joi.string().isoDate().required()
+});
+
+// Environment variables
+const {
+    RDS_SECRET_ARN,
+    AWS_REGION,
+} = process.env;
+
+// AWS clients
+const secretsManager = new AWS.SecretsManager({ region: AWS_REGION });
+
+/**
+ * Lambda handler for processing adoption history messages from SQS
+ */
+exports.handler = async (event, context) => {
+    console.log('Processing adoption history messages:', JSON.stringify(event, null, 2));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Log Injection occurs when untrusted user input is directly written to log files without proper sanitization. This can allow attackers to manipulate log entries, potentially leading to security issues like log forging or cross-site scripting. To prevent this, always sanitize user input using `encodeURIComponent()` or `DOMPurify.sanitize()` before logging.  Learn more - https://cwe.mitre.org/data/definitions/117.html